src/logging.rs

```diff
@@ -78,12 +78,29 @@ fn is_valid_field(input: &str) -> bool {
     true
 }
 
-fn add_field_and_payload(data: &mut String, field: &str, payload: &str) {
+fn add_field_and_payload(data: &mut Vec<u8>, field: &str, payload: &str) {
     if is_valid_field(field) {
-        let field_payload = format!("{}={}\n", field, payload);
-        data.push_str(&field_payload);
+        if payload.contains('\n') {
+            // If the payload contains a newline we need to encode the payload in a more intricate
+            // way, see https://github.com/systemd/systemd/blob/d65c5d04f9c6d6c943e67e677161caed782fe7a7/src/libsystemd/sd-journal/journal-send.c#L250
```

Comment:
I think that the canonical URL for this is https://systemd.io/JOURNAL_NATIVE_PROTOCOL/.

Reply:
I'll update the link.

Reply:
I've added a doc comment to this function and linked the protocol docs there.